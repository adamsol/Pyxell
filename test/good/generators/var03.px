
func counter(n: Int): Int... def
    print true
    for i in 1..n do
        yield i
        print i

r = counter(5)
print false

print [for v, i in r, 0...3 yield v]
# 4 is lost here, Python behaves the same way.
print [...r]

r = counter(5)

print [for i, v in 0...3, r yield v]
# But with another order of iterables, 4 is consumed properly.
print [...r]
