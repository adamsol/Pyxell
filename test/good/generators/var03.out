false
true
1
2
3
[1, 2, 3]
4
5
[5]
true
1
2
[1, 2, 3]
3
4
5
[4, 5]
